doc/Command-Reference.md

```diff
@@ -1734,6 +1737,173 @@ This command is used to remove particular IPv4 or IPv6 BGP neighbor configuratio
 
 Go Back To [Beginning of the document](#) or [Beginning of this section](#bgp)
 
+
+## Component Firmware
+
+### Component Firmware show commands
+
+**show platform firmware**
+
+This command displays platform components firmware status information.
+
+- Usage:
+```bash
+show platform firmware
+```
+
+- Example:
+```bash
+root@sonic:/home/admin# show platform firmware
+Chassis    Module    Component    Version                  Description
+---------  --------  -----------  -----------------------  ---------------------------------------
+Chassis1   N/A       BIOS         0ACLH004_02.02.007_9600  BIOS - Basic Input/Output System
+                     CPLD         5.3.3.1                  CPLD - includes all CPLDs in the switch
+```
+
+### Component Firmware config command
+
+**config platform firmware install**
+
+This command is used to install a platform component firmware.  
+Both modular and non modular chassis platforms are supported.
+
+- Usage:
+```bash
+config platform firmware install chassis component <component_name> fw <fw_path> [-y|--yes]
+config platform firmware install module <module_name> component <component_name> fw <fw_path> [-y|--yes]
+```
+
+- Example:
+```bash
+root@sonic:/home/admin# config platform firmware install chassis component BIOS fw /etc/mlnx/fw/sn3800/chassis1/bios.bin
+New firmware will be installed, continue? [y/N]: y
+Installing firmware:
+    /etc/mlnx/fw/sn3800/chassis1/bios.bin
+
+root@sonic:/home/admin# config platform firmware install module Module1 component BIOS fw http://mellanox.com/fw/sn3800/module1/bios.bin
+New firmware will be installed, continue? [y/N]: y
+Downloading firmware:
+    [##################################################]  100%
+Installing firmware:
+    /tmp/bios.bin
+```
+
+Supported options:
+1. -y|--yes - automatic yes to prompts. Assume "yes" as answer to all prompts and run non-interactively
+
```

Comment:
if a reload is required, does the api inform it somehow to the user? if so, we need to make it clear in the documentation.

Reply:
@liat-grozovik if a reload is required the API may inform user thru the stdout. It's a vendor specific implementation.